src/include/parser/expression/abstract_expression.h

```diff
@@ -26,6 +28,18 @@ class AbstractExpression {
   AbstractExpression(const ExpressionType expression_type, const type::TypeId return_value_type,
                      std::vector<std::shared_ptr<AbstractExpression>> &&children)
       : expression_type_(expression_type), return_value_type_(return_value_type), children_(std::move(children)) {}
+  /**
+   * Instantiates a new abstract expression with alias used for select statement column references.
+   * @param expression_type what type of expression we have
+   * @param return_value_type the type of the expression's value
+   * @param alias alias of the column (used in column value expression)
+   * @param children the list of children for this node
+   */
+  AbstractExpression(const ExpressionType expression_type, const type::TypeId return_value_type, const char *alias,
+                     std::vector<std::shared_ptr<AbstractExpression>> &&children)
+      : expression_type_(expression_type), return_value_type_(return_value_type), children_(std::move(children)) {
+    if (alias != nullptr) alias_ = std::string(alias);
```

Comment:
postgresparser parse the text with thrid party component and initialize an expression with whatever in the parse node. I can check if alias is nullptr in the postgres parser and call corresponding constructor depending on whether alias is there to avoid the above

Reply:
I see, then yea I would handle that logic in the parser and call the corresponding constructor to make the difference between the constructors explicit.